user_group/views.py

```diff
@@ -66,3 +68,30 @@ def finalize_response(self, request, response, *args, **kwargs):
 
     def get_queryset(self):
         return GroupMembership.get_for(self.request.user)
+
+
+class UserGroupUserSearchViewSet(viewsets.ViewSet):
+    permission_classes = [permissions.IsAuthenticated]
+
+    def list(self, request, version=None):
+        query = request.query_params.get('query', '').strip().lower()
+        if not query:
+            return response.Response(
+                {'errors': ['Empty query']},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        users = User.objects.filter(
+            Q(username__icontains=query) |
+            Q(first_name__icontains=query) |
+            Q(last_name__icontains=query) |
+            Q(email__icontains=query)
+        ).\
+            annotate(title=F('username'), type=Value('user', CharField())).\
+            values('id', 'title', 'type')
+        user_groups = UserGroup.objects.filter(title__icontains=query).\
+            annotate(type=Value('user_group', CharField())).\
+            values('id', 'title', 'type')
+        items = list(users) + list(user_groups)
+        return response.Response(
```

Comment:
For consistent data structure expected by client, the result should be in the paginated format. Use `self.get_paginated_response(items)`.

Reply:
Actually if we union the `users` and `user_groups` into single query set, we can use serializer and let django handle the pagination and response format.